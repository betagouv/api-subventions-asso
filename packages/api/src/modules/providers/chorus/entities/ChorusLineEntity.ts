import { ObjectId } from "mongodb";
import { ParserInfo } from "../../../../@types";
import { ExcelDateToJSDate } from "../../../../shared/helpers/ParserHelper";
import IChorusIndexedInformations from "../@types/IChorusIndexedInformations";

export default class ChorusLineEntity {
    public provider = "Chorus";

    public static indexedInformationsPath: {[key: string]: ParserInfo} = {
        ej: {
            path: ["N° EJ"],
        },
        siret: { path: ['Code taxe 1'] },
        codeBranche: { path: ['Branche CODE'] },
        branche: { path: ['Branche'] }, 
        activitee: { path: ["Référentiel de programmation"]},
        numeroDemandePayment: { path: ["N° DP"]},
        numeroTier: { path: ["Fournisseur payé (DP)"]},
        centreFinancier: { path: ["Centre financier"]},
        codeCentreFinancier: { path: ["Centre financier CODE"]},
        domaineFonctionnel: { path: ["Domaine fonctionnel"]},
        codeDomaineFonctionnel: { path: ["Domaine fonctionnel CODE"]},
        amount: { 
            path: [["EUR", "Montant payé"]],
            adapter: (value) => {
                if (!value || typeof value === "number") return value;
                
                return parseFloat(value.replace("\r", "").replace(" ", '').replace(",", "."))
            }
        }, 
        dateOperation: { 
            path: ["Date de dernière opération sur la DP"],
            adapter: (value) => {
                if (!value) return value;
                if (value != parseInt(value, 10).toString()) {
                    const [day, month, year] = value.split(/[/.]/).map(v => parseInt(v, 10));
                    return new Date(Date.UTC(year, month - 1, day));
                }

                return ExcelDateToJSDate(parseInt(value, 10));
            }
        }, 
    }

    constructor(
        public uniqueId: string,
        public indexedInformations: IChorusIndexedInformations,
        public data: unknown,
        public _id?: ObjectId
    ) {}
}
