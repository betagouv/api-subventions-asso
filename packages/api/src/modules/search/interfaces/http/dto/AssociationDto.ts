import DemandeSubvention from "@api-subventions-asso/dto/search/DemandeSubventionDto";
import Versement from "@api-subventions-asso/dto/search/VersementDto";
import Association from "../../../../associations/@types/Association";
import Etablissement from "../../../../etablissements/@types/Etablissement";

export default interface AssociationDto extends Association {
    etablissements: ({ demandes_subventions: DemandeSubvention[] | null, versements: Versement[] | null } & Etablissement)[] | null,
}