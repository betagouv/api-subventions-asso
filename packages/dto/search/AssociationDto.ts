import { ProviderValues } from "../shared/ProviderValue";
import { Rna } from "../shared/Rna";
import { Siren } from "../shared/Siren";
import { Siret } from "../shared/Siret";

export default interface AssociationDto {
    siren?: ProviderValues<Siren>,
    rna?: ProviderValues<Rna>,
    nic_siege?: ProviderValues<string>,
    categorie_juridique?: ProviderValues<string>,
    denomination: ProviderValues<string>,
    date_creation?: ProviderValues<Date>,
    date_modification?: ProviderValues<Date>,
    objet_social?: ProviderValues<string>;
    code_objet_social_1?: ProviderValues<string>;
    code_objet_social_2?: ProviderValues<string>;
    etablisements_siret?: ProviderValues<Siret[]>;
    adresse_siege?: ProviderValues<{ // via rna et ou osiris
        numero?: string,
        type_voie?: string,
        voie?: string,
        code_postal?: string,
        commune?: string,
    }>,
    federation?: ProviderValues<string>,
    licencies?: { // Uniquement les asso sportive
        total?: ProviderValues<number>,
        hommes?: ProviderValues<number>,
        femmes?: ProviderValues<number>,
    },
    benevoles?: {
        nombre?: ProviderValues<number>,
        ETPT?: ProviderValues<number>,
    },
    salaries?: {
        nombre?: ProviderValues<number>,
        cdi?: ProviderValues<number>,
        cdiETPT?: ProviderValues<number>,
        cdd?: ProviderValues<number>,
        cddETPT?: ProviderValues<number>,
        emploisAides?: ProviderValues<number>,
        emploisAidesETPT?: ProviderValues<number>
    },
    volontaires?: {
        nombre?: ProviderValues<number>,
        ETPT?: ProviderValues<number>,
    }
    // etablissements: ({ demandes_subventions: DemandeSubvention[] | null } & Etablissement)[] | null,
}