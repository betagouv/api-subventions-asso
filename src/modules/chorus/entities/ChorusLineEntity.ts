import { ObjectId } from "mongodb";
import { ParserInfo } from "../../../@types/ParserInfo";
import IChorusIndexedInformations from "../@types/IChorusIndexedInformations";

export default class ChorusLineEntity {
    public provider = "Chorus";

    public static indexedInformationsPath: {[key: string]: ParserInfo} = {
        ej: {
            path: ["N° EJ"],
        },
        siret: { path: ['Code taxe 1'] },
        compte: { path: ['Compte général'] }, 
        amount: { 
            path: ["EUR"],
            adapter: (value) => {
                if (!value) return value;
                return parseFloat(value.replace("\r", "").replace(" ", ''))
            }
        }, 
        dateOperation: { 
            path: ["Date de dernière opération sur la DP"],
            adapter: (value) => {
                if (!value) return value;

                const [day, month, year] = value.split('.').map(v => parseInt(v, 10));
                return new Date(year, month - 1, day);
            }
        }, 
    }

    constructor(
        public indexedInformations: IChorusIndexedInformations,
        public data: unknown,
        public _id?: ObjectId
    ) {}
}
