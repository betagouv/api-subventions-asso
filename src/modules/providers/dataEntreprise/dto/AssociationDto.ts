export default interface AssociationDto {
    association: {
        id: number,
        is_waldec: string,
        id_association: string,
        id_ex_association: string,
        siret?: string,
        numero_reconnaissance_utilite_publique: string,
        code_gestion: string,
        date_creation: string,
        date_derniere_declaration: string,
        date_publication_creation: string,
        date_declaration_dissolution?: string,
        nature: string
        groupement: string,
        titre: string,
        titre_court: string,
        objet: string,
        objet_social1: string,
        objet_social2: string,
        l1_adresse_import?: string,
        l2_adresse_import?: string,
        l3_adresse_import?: string,
        adresse_siege: string,
        adresse_numero_voie: string,
        adresse_repetition?: string,
        adresse_type_voie: string,
        adresse_libelle_voie: string,
        adresse_distribution:string,
        adresse_code_insee: string,
        adresse_code_postal: string,
        adresse_libelle_commune: string,
        adresse_gestion_nom?: string,
        adresse_gestion_format_postal: string,
        adresse_gestion_geo?: string,
        adresse_gestion_libelle_voie: string,
        adresse_gestion_distribution?: string,
        adresse_gestion_code_postal: string,
        adresse_gestion_acheminement: string,
        adresse_gestion_pays:string,
        dirigeant_civilite: string,
        telephone?: string,
        site_web?: string,
        email?: string,
        autorisation_publication_web: string,
        observation: string,
        position_activite: string,
        derniere_maj: string,
        created_at: string,
        updated_at: string
    }
}