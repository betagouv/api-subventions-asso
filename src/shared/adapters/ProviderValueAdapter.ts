import ProviderValue from "../../@types/ProviderValue";

export default class ProviderValueAdapter {
    static toProviderValue<T>(value: T, provider: string, updated_at: Date): ProviderValue<T> {
        return {
            value,
            provider,
            last_update: updated_at,
            type: typeof value
        }
    }
}