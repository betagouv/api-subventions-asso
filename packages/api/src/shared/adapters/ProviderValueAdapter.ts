import { ProviderValue, ProviderValues } from "@api-subventions-asso/dto";

export default class ProviderValueAdapter {

    static isProviderValue(value: unknown): value is ProviderValue<unknown> {
        return !!value 
                && "value" in (value as ProviderValue<unknown>) 
                && "provider" in (value as ProviderValue<unknown>);
    }

    static isProviderValues(values: unknown) : values is ProviderValues {
        return Array.isArray(values) 
            && !!values.filter(v => v).length 
            && values.every(v => ProviderValueAdapter.isProviderValue(v));
    }

    static toProviderValue<T>(value: T, provider: string, updated_at: Date): ProviderValue<T> {
        return { 
            value,
            provider,
            last_update: updated_at,
            type: typeof value
        }
    }

    static toProviderValues<T>(value: T, provider: string, updated_at: Date): ProviderValues<T> {
        return [{
            value,
            provider,
            last_update: updated_at,
            type: typeof value
        }]
    }
}