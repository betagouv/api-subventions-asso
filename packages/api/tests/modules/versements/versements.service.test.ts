import chorusService from "../../../src/modules/providers/chorus/chorus.service";
import ChorusLineEntity from "../../../src/modules/providers/chorus/entities/ChorusLineEntity";
import AssociationDto from "../../../src/modules/search/interfaces/http/dto/AssociationDto";
import EtablissementDto from "../../../src/modules/search/interfaces/http/dto/EtablissmentDto";
import versementsService from "../../../src/modules/versements/versements.service";
import ProviderValueAdapter from "../../../src/shared/adapters/ProviderValueAdapter";

describe("VersementService", () => {
    const now = new Date();
    const toPVs = <T=unknown>(value: T, provider = "Chorus") => ProviderValueAdapter.toProviderValues(value, provider, now);
    const toPV = (value: unknown, provider = "Chorus") => ProviderValueAdapter.toProviderValue(value, provider, now);
    
    describe("aggregateVersementsByAssoSearch", () => {
        const asso = {
            siren: toPVs("100000000"),
            etablissements: [
                {
                    siret: toPVs("10000000000000"),
                    demandes_subventions: [
                        {
                            ej: toPV("1000000000")
                        }
                    ]
                }
            ]
        } as unknown as AssociationDto;

        beforeEach(async () => {
            jest.spyOn(chorusService, "sirenBelongAsso").mockImplementation(() => Promise.resolve(true))
            await chorusService.addChorusLine(new ChorusLineEntity("FAKE_ID", {
                siret: "10000000000000",
                ej: "1000000000",
                amount: 1000,
                dateOperation: now,
                codeBranche: "Z004",
                branche: "BRANCHE",
                compte: "COMPTE",
                typeOperation: "ZSUB"
            }, {}));
        })

        it("should be aggregate versements", () => {
            expect(versementsService.aggregateVersementsByAssoSearch(asso)).resolves.toMatchObject({
                siren: toPVs("100000000"),
                versements:  [
                    {
                        siret: toPV("10000000000000"),
                        ej: toPV("1000000000"),
                        amount: toPV(1000),
                        dateOperation: toPV(now),
                        codeBranche: toPV("Z004"),
                        branche: toPV("BRANCHE"),
                        compte: toPV("COMPTE"),
                        type: toPV("ZSUB")
                    }
                ],
                etablissements: [
                    {
                        siret: toPVs("10000000000000"),
                        versements: [{
                            siret: toPV("10000000000000"),
                            ej: toPV("1000000000"),
                            amount: toPV(1000),
                            dateOperation: toPV(now),
                            codeBranche: toPV("Z004"),
                            compte: toPV("COMPTE")
                        }],
                        demandes_subventions: [
                            {
                                ej: toPV("1000000000"),
                                versements:  [{
                                    siret: toPV("10000000000000"),
                                    ej: toPV("1000000000"),
                                    amount: toPV(1000),
                                    dateOperation: toPV(now),
                                    codeBranche: toPV("Z004"),
                                    compte: toPV("COMPTE")
                                }]
                            }
                        ]
                    }
                ]
            })
        })

        it("should be not aggregate versements", () => {
            expect(versementsService.aggregateVersementsByAssoSearch({...asso, siren: toPVs("100000080")})).resolves.toMatchObject({
                siren: toPVs("100000080"),
                versements: [],
                etablissements: [
                    {
                        siret: toPVs("10000000000000"),
                        versements: [],
                        demandes_subventions: [
                            {
                                ej: toPV("1000000000"),
                                versements:  []
                            }
                        ]
                    }
                ]
            })
        });
    });


     
    describe("aggregateVersementsByEtablissementSearch", () => {
        const etablissementDto = {
            siret: toPVs("10000000000000"),
            demandes_subventions: [
                {
                    ej: toPV("1000000000")
                }
            ]
        } as unknown as EtablissementDto;

        beforeEach(async () => {
            jest.spyOn(chorusService, "sirenBelongAsso").mockImplementation(() => Promise.resolve(true))
            await chorusService.addChorusLine(new ChorusLineEntity("FAKE_ID", {
                siret: "10000000000000",
                ej: "1000000000",
                amount: 1000,
                dateOperation: now,
                codeBranche: "Z004",
                branche: "BRANCHE",
                compte: "COMPTE",
                typeOperation: "ZSUB"
            }, {}));
        })

        it("should be aggregate versements", () => {
            expect(versementsService.aggregateVersementsByEtablissementSearch(etablissementDto)).resolves.toMatchObject({
                siret: toPVs("10000000000000"),
                versements: [{
                    siret: toPV("10000000000000"),
                    ej: toPV("1000000000"),
                    amount: toPV(1000),
                    dateOperation: toPV(now),
                    codeBranche: toPV("Z004"),
                    compte: toPV("COMPTE")
                }],
                demandes_subventions: [
                    {
                        ej: toPV("1000000000"),
                        versements:  [{
                            siret: toPV("10000000000000"),
                            ej: toPV("1000000000"),
                            amount: toPV(1000),
                            dateOperation: toPV(now),
                            codeBranche: toPV("Z004"),
                            compte: toPV("COMPTE")
                        }]
                    }
                ]
            })
        })

        it("should be not aggregate versements", () => {
            expect(versementsService.aggregateVersementsByEtablissementSearch({...etablissementDto, siret: toPVs("10000000008000")})).resolves.toMatchObject({
                siret: toPVs("10000000008000"),
                versements: [],
                demandes_subventions: [
                    {
                        ej: toPV("1000000000"),
                        versements:  []
                    }
                ]
            })
        });
    });
});