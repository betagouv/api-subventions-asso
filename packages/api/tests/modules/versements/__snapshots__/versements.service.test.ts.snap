// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersementService aggregateVersementsByAssoSearch should aggregate versements 1`] = `
Object {
  "etablissements": Array [
    Object {
      "demandes_subventions": Array [
        Object {
          "ej": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "1000000000",
          },
          "versementKey": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "1000000000",
          },
          "versements": Array [
            Object {
              "activitee": undefined,
              "amount": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "number",
                "value": 1000,
              },
              "branche": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "BRANCHE",
              },
              "centreFinancier": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "CENTRE_FINANCIER",
              },
              "codeBranche": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "Z004",
              },
              "compte": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "COMPTE",
              },
              "dateOperation": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "object",
                "value": 2022-01-01T00:00:00.000Z,
              },
              "domaineFonctionnel": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "DOMAINE_FONCTIONNEL",
              },
              "ej": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "1000000000",
              },
              "numeroDemandePayment": undefined,
              "numeroTier": undefined,
              "siret": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "10000000000000",
              },
              "type": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "ZSUB",
              },
              "versementKey": Object {
                "last_update": 2022-01-01T00:00:00.000Z,
                "provider": "Chorus",
                "type": "string",
                "value": "1000000000",
              },
            },
          ],
        },
        Object {
          "codePoste": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "J00034",
          },
          "versementKey": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "J00034",
          },
          "versements": Array [],
        },
      ],
      "siret": Array [
        Object {
          "last_update": 2022-01-01T00:00:00.000Z,
          "provider": "Chorus",
          "type": "string",
          "value": "10000000000000",
        },
      ],
      "versements": Array [
        Object {
          "activitee": undefined,
          "amount": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "number",
            "value": 1000,
          },
          "branche": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "BRANCHE",
          },
          "centreFinancier": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "CENTRE_FINANCIER",
          },
          "codeBranche": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "Z004",
          },
          "compte": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "COMPTE",
          },
          "dateOperation": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "object",
            "value": 2022-01-01T00:00:00.000Z,
          },
          "domaineFonctionnel": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "DOMAINE_FONCTIONNEL",
          },
          "ej": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "1000000000",
          },
          "numeroDemandePayment": undefined,
          "numeroTier": undefined,
          "siret": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "10000000000000",
          },
          "type": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "ZSUB",
          },
          "versementKey": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "1000000000",
          },
        },
      ],
    },
  ],
  "siren": Array [
    Object {
      "last_update": 2022-01-01T00:00:00.000Z,
      "provider": "Chorus",
      "type": "string",
      "value": "100000000",
    },
  ],
  "versements": Array [
    Object {
      "activitee": undefined,
      "amount": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "number",
        "value": 1000,
      },
      "branche": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "BRANCHE",
      },
      "centreFinancier": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "CENTRE_FINANCIER",
      },
      "codeBranche": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "Z004",
      },
      "compte": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "COMPTE",
      },
      "dateOperation": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "object",
        "value": 2022-01-01T00:00:00.000Z,
      },
      "domaineFonctionnel": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "DOMAINE_FONCTIONNEL",
      },
      "ej": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "1000000000",
      },
      "numeroDemandePayment": undefined,
      "numeroTier": undefined,
      "siret": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "10000000000000",
      },
      "type": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "ZSUB",
      },
      "versementKey": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "1000000000",
      },
    },
  ],
}
`;

exports[`VersementService aggregateVersementsByAssoSearch should not aggregate versements 1`] = `
Object {
  "etablissements": Array [
    Object {
      "demandes_subventions": Array [
        Object {
          "ej": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "1000000000",
          },
          "versementKey": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "1000000000",
          },
          "versements": Array [],
        },
        Object {
          "codePoste": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "J00034",
          },
          "versementKey": Object {
            "last_update": 2022-01-01T00:00:00.000Z,
            "provider": "Chorus",
            "type": "string",
            "value": "J00034",
          },
          "versements": Array [],
        },
      ],
      "siret": Array [
        Object {
          "last_update": 2022-01-01T00:00:00.000Z,
          "provider": "Chorus",
          "type": "string",
          "value": "10000000000000",
        },
      ],
      "versements": Array [],
    },
  ],
  "siren": Array [
    Object {
      "last_update": 2022-01-01T00:00:00.000Z,
      "provider": "Chorus",
      "type": "string",
      "value": "100000080",
    },
  ],
  "versements": Array [],
}
`;

exports[`VersementService aggregateVersementsByEtablissementSearch should aggregate versements 1`] = `Promise {}`;

exports[`VersementService aggregateVersementsByEtablissementSearch should not aggregate versements 1`] = `
Object {
  "demandes_subventions": Array [
    Object {
      "ej": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "1000000000",
      },
      "versementKey": Object {
        "last_update": 2022-01-01T00:00:00.000Z,
        "provider": "Chorus",
        "type": "string",
        "value": "1000000000",
      },
      "versements": Array [],
    },
  ],
  "siret": Array [
    Object {
      "last_update": 2022-01-01T00:00:00.000Z,
      "provider": "Chorus",
      "type": "string",
      "value": "10000000008000",
    },
  ],
  "versements": Array [],
}
`;
